src/Authentication.Abstractions/AzureEnvironmentConstants.cs

```diff
@@ -86,6 +95,7 @@ public static class AzureEnvironmentConstants
         public const string AzureActiveDirectoryEndpoint = "https://login.microsoftonline.com/";
         public const string ChinaActiveDirectoryEndpoint = "https://login.chinacloudapi.cn/";
         public const string USGovernmentActiveDirectoryEndpoint = "https://login.microsoftonline.us/";
+        [Obsolete("Microsoft Cloud Germany was closed on October 29th, 2021.")]
```

Comment:
Why is the constant marked as `Obsolete` instead of being removed if they are not accessible anymore (https://github.com/Azure/azure-cli/issues/23775)?

Reply:
This repo contains code that is depended by all the azure powershell modules. It's a general practice to maintain backwards compatibility here. Removing this property would be a binary breaking change.